sdk/core/src/type_map_context.rs

```diff
@@ -0,0 +1,36 @@
+use crate::OverridableContext;
+use std::sync::Arc;
+
+/// Execution context.
+///
+/// The `Context` is modeled as an overridable chain. In other words, the `Context` allows to
+/// override any value without losing the original one. The resulting type map is the union of the
+/// type map of every chained `Context`, giving precedence to the last inserted one. As a simple
+/// example suppose to have a `Context` with the key value of `Struct1` -> `Hello` and `Struct2` ->
+/// `World`. Let's call it `C1`.
+/// If we chain (using the `from_previous_context` function) this `Context` we get another
+/// `Context`, called `C2`.
+/// Calling `get` on C2 is now the same of calling `get` on `C1`: `Struct2` gives `World`. But now
+/// we can replace `Struct2` with `Solar system` without losing the original `C1`'s `World`. If we
+/// do that, `C2.get::<Struct2>()` will give you `Solar system`, while `C1.get::<Struct2>()` will
+/// give you `World`.
+pub trait TypeMapContext {
```

Comment:
Why is this trait needed? Is this just to ensure that `Context` and `OverrideableContext` have the same methods? We don't seem to ever require that in code and so this feels like it just complicates things.